actors/evm/src/interpreter/memory.rs

```diff
@@ -16,12 +16,15 @@ impl Default for Memory {
 
 impl Memory {
     #[inline]
+    /// Resizes memory to `size` length, reserving extra WASM pages as-needed.
+    /// TODO this should be renamed resize since it can also shrink memory.
     pub fn grow(&mut self, size: usize) {
         let cap = self.0.capacity();
         if size > cap {
             let required_pages = (size + PAGE_SIZE - 1) / PAGE_SIZE;
             self.0.reserve((PAGE_SIZE * required_pages) - self.0.len());
         }
+        debug_assert_eq!(size % 32, 0, "MSIZE depends on memory length being multiple of 32");
```

Comment:
There are too many cases to cover with unit tests practically that grows memory size, so this will fail on any test run that breaks this invariant. We could re-write this as well to ensure this always grows memory by multiples of 32 bytes though.

Reply:
This is a good place for a debug assert given that this is a precondition of the method.